src/components/navigation/sub-tabs-panel.tsx

```diff
@@ -0,0 +1,76 @@
+import React from "react";
+import { observer } from "mobx-react";
+import { Tab, TabList, TabPanel, Tabs } from "react-tabs";
+import classNames from "classnames";
+import { useAppConfig } from "../../hooks/use-stores";
+import { ISubTabSpec, NavTabModelType } from "../../models/view/nav-tabs";
+
+import "./sub-tabs-panel.sass";
+
+interface IProps {
+  tabSpec: NavTabModelType;
+  renderSubTabPanel: (subTab: ISubTabSpec) => JSX.Element;
+  tabsExtraClassNames?: classNames.Argument,
+  onSelect?: (tabIdx: number) => void,
+  // This will make the tabs a controlled component
+  selectedIndex?: number,
+}
+
+const kHeaderHeight = 55;
+const kWorkspaceContentMargin = 4;
+const kNavTabHeight = 34;
+const kTabSectionBorderWidth = 2;
+
+export const SubTabsPanel: React.FC<IProps> = observer(function SubTabsPanel(
+    { tabSpec, renderSubTabPanel, tabsExtraClassNames, onSelect, selectedIndex }) {
+  const appConfigStore = useAppConfig();
+  const navTabSpec = appConfigStore.navTabs.getNavTabSpec(tabSpec.tab);
+  const subTabs = tabSpec.subTabs;
+  const hasSubTabs = subTabs.length > 1;
+
+  const vh = window.innerHeight;
+  const headerOffset = hasSubTabs
+                        ? kHeaderHeight + (2 * (kWorkspaceContentMargin + kNavTabHeight + kTabSectionBorderWidth))
+                        : kHeaderHeight + kNavTabHeight + (2 * (kWorkspaceContentMargin + kTabSectionBorderWidth));
```

Comment:
I just copied this code without figuring it out. Since you suggested an alternative I dug into it more. If we are going to stick with this non-css approach my preference is something explicit like:
```
  const headerOffset =
    kHeaderHeight +
    // Main Tabs
    kWorkspaceContentMargin + kNavTabHeight + kTabSectionBorderWidth +
    // Sub Tabs: even without sub tabs content margin and section border is present
    kWorkspaceContentMargin +
    (hasSubTabs ? kNavTabHeight : 0) +
    kTabSectionBorderWidth;
```

But really we should figure out how to do this with CSS so aren't hard coding heights of things. Since folks are waiting on me to merge this as is, and maybe if I can find time I'll try to make this use CSS instead.